src/cdk-experimental/listbox/listbox.ts

```diff
@@ -8,182 +8,146 @@
 
 import {
   AfterContentInit,
+  ChangeDetectorRef,
   ContentChildren,
   Directive,
   ElementRef,
-  EventEmitter,
   forwardRef,
   Inject,
   Input,
   OnDestroy,
-  OnInit,
   Optional,
   Output,
   QueryList,
 } from '@angular/core';
 import {ActiveDescendantKeyManager, Highlightable, ListKeyManagerOption} from '@angular/cdk/a11y';
 import {DOWN_ARROW, ENTER, LEFT_ARROW, RIGHT_ARROW, SPACE, UP_ARROW} from '@angular/cdk/keycodes';
 import {BooleanInput, coerceArray, coerceBooleanProperty} from '@angular/cdk/coercion';
-import {SelectionChange, SelectionModel} from '@angular/cdk/collections';
-import {defer, merge, Observable, Subject} from 'rxjs';
-import {startWith, switchMap, takeUntil} from 'rxjs/operators';
+import {SelectionModel} from '@angular/cdk/collections';
+import {BehaviorSubject, defer, merge, Observable, Subject} from 'rxjs';
+import {filter, mapTo, startWith, switchMap, takeUntil} from 'rxjs/operators';
 import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
 import {Directionality} from '@angular/cdk/bidi';
 import {CDK_COMBOBOX, CdkCombobox} from '@angular/cdk-experimental/combobox';
 
-let nextId = 0;
-let listboxId = 0;
+/** The next id to use for the CdkOption directive. */
+let nextOptionId = 0;
 
-export const CDK_LISTBOX_VALUE_ACCESSOR = {
-  provide: NG_VALUE_ACCESSOR,
-  useExisting: forwardRef(() => CdkListbox),
-  multi: true,
-};
+/** The next id to use for the CdkListbox directive. */
+let nextListboxId = 0;
```

Comment:
I don't think we need two separate counters for IDs, we can use one for the listbox and options.